### DISCUSSION TITLE
"offline" score submission support

### DISCUSSION THREAD

**Discussion by AnasElgamed** [ppy/osu#19912] [Ideas]
In lazer you can only play offline as a guest 
but in stable osu you can play offline and when you are back online the scores are uploaded (as long as you haven't closed the game)
it would be really nice if you add something similar to what happens in stable osu 
and if the scores are saved even if you quit the game it would be even better

**Comment by peppy** [comment]
What is your use case?

**Reply by AnasElgamed** [reply]
The internet in my country is quite bad and sometimes I become without internet (due to technical issuses and maintenance  as they say ) so offline mode is really helpful for me

**Reply by Feodor0090** [reply]
I would like to mention mine - i often play on laptop when i'm on train or something like that.

I can login normally when i'm leaving home and switch laptop to sleep mode, then play offline, put laptop to sleep mode again and some time after (when i arrive) connect to the internet again. Would be nice if the game keep all the scores/replays in ram (and show in notifications something like "XYZ scores are waiting to be submitted") and send them once you are online.

The only problem here that (at least on my system) you need some luck to keep the game opened more than several hours in a row due to random segfaults from BASS library, but that's another issue.

**Reply by peppy** [reply]
Unfortunately I don't think lazer will support the latter use case. We'll do our best to make it resilient against connection failures, though.

It will 100% support saving such gameplay locally as well.
